kroxylicious-integration-tests/src/test/java/io/kroxylicious/proxy/TlsIT.java

```diff
@@ -411,6 +422,35 @@ void downstream_SuccessfulTlsWithProtocolsAllowed(KafkaCluster cluster) {
         }
     }
 
+    @Test
+    void downstream_UnrecognizedSniHostNameClosesConnection(KafkaCluster cluster) {
+        var duffBootstrap = "bootstrap." + IntegrationTestInetAddressResolverProvider.generateFullyQualifiedDomainName("duff") + ":" + SNI_BOOTSTRAP_ADDRESS.port();
+
+        var builder = new ConfigurationBuilder()
+                .addToVirtualClusters(baseVirtualClusterBuilder(cluster, "demo")
+                        .addToGateways(defaultSniHostIdentifiesNodeGatewayBuilder(SNI_BOOTSTRAP_ADDRESS, SNI_BROKER_ADDRESS_PATTERN)
+                                .withNewTls()
+                                .withNewKeyStoreKey()
+                                .withStoreFile(downstreamCertificateGenerator.getKeyStoreLocation())
+                                .withNewInlinePasswordStoreProvider(downstreamCertificateGenerator.getPassword())
+                                .endKeyStoreKey()
+                                .endTls()
+                                .build())
+                        .build());
+
+        try (var tester = kroxyliciousTester(builder);
+                var testClient = tester.simpleTestClient(duffBootstrap, true)) {
+
+            // The request itself is unimportant, it won't actually reach the server
+            var request = new Request(ApiKeys.API_VERSIONS, ApiKeys.API_VERSIONS.latestVersion(), null, new ApiVersionsRequestData());
+            var response = testClient.get(request);
+            assertThat(response)
```

Comment:
If you take the fix away, you'll find that the future fails to complete.  This is because the connection is left open.